app/assets/javascripts/common/reports.js

```diff
@@ -662,83 +662,69 @@ Reports = function ({
+        ? dateKeys[defaultMonthsRequired - 1] // month 6
         : mostRecentDecemberKey; // december
 
+    const monthThreeIndex = dateKeys.indexOf(monthThreeDateKey);
+
     const monthThreeValue = periodValues[monthThreeDateKey];
-    const indexOfMonthThreeInDateKeys = dateKeysArray.indexOf(monthThreeDateKey);
-    const monthTwoValue = periodValues[dateKeysArray[indexOfMonthThreeInDateKeys - 1]];
-    const monthOneValue = periodValues[dateKeysArray[indexOfMonthThreeInDateKeys - 2]];
+    const monthTwoValue = periodValues[dateKeys[monthThreeIndex - 1]];
+    const monthOneValue = periodValues[dateKeys[monthThreeIndex - 2]];
 
     const sumValues = monthOneValue + monthTwoValue + monthThreeValue;
     const threeMonthAverage = sumValues === 0 ? 0 : sumValues / 3;
+    
+    const monthThreeIndexOfYear = monthIndexFromDateKey(monthThreeDateKey);
+    const improvementRatio = relativeImprovementRatio(monthThreeIndexOfYear);
 
-    const monthThreeIndexOfYear = monthIndexFromDateString(monthThreeDateKey);
     return {
       threeMonthAverage,
-      monthThreeIndexOfYear,
+      improvementRatio,
     };
   }
 
-  function monthIndexFromDateString(dateString) {
-    const [month, year] = dateString.split("-");
-    const months = [
-      "jan",
-      "feb",
-      "mar",
-      "apr",
-      "may",
-      "jun",
-      "jul",
-      "aug",
-      "sep",
-      "oct",
-      "nov",
-      "dec",
-    ];
-
-    return months.indexOf(month.toLowerCase());
+  function monthIndexFromDateKey(dateString) {
+    return new Date(dateString).getMonth();
   }
 
-  function calculateGoalUpwards(monthValue, improvementRatio) {
-    const goal = monthValue + (100 - monthValue) * improvementRatio;
-    return Math.ceil(goal);
+  function relativeImprovementRatio(monthThreeIndex) {
+    const annualRelativeImprovement = 0.1; // 10%
+    const monthlyRelativeImprovement = annualRelativeImprovement / 12;
+    const monthsRemainingForYear = monthThreeIndex === 11 ? 12 : 11 - monthThreeIndex; // dec = 12, jan = 11... nov = 1
+    return monthlyRelativeImprovement * monthsRemainingForYear;
   }
 
-  function calculateGoalDownwards(monthValue, improvementRatio) {
-    const goal = monthValue - monthValue * improvementRatio;
-    return Math.floor(goal);
+  function calculateGoalUpwards(threeMonthAverage, improvementRatio) {
+    const goal = threeMonthAverage + (100 - threeMonthAverage) * improvementRatio;
+    return Math.ceil(goal);
   }
 
-  function relativeImprovementRatio(goalMonthIndex) {
-    const defaultRelativeImprovementPercentage = 10;
-    const improvementPercentagePerMonth =
-      defaultRelativeImprovementPercentage / 100 / 12;
-    const monthsRemainingInYear = 12 - (goalMonthIndex % 11); // dec is full year (0 index)
-    return improvementPercentagePerMonth * monthsRemainingInYear;
+  function calculateGoalDownwards(threeMonthAverage, improvementRatio) {
+    const goal = threeMonthAverage - threeMonthAverage * improvementRatio;
```

Comment:
[nitpick] It's a good practice to add brackets around the expression that is to be evaluated first
```suggestion
    const goal = threeMonthAverage - (threeMonthAverage * improvementRatio);
```